### PR TITLE
feat: substition of class-validator to zod in dtos from users and results

### PR DESCRIPTION
**Pull request by XVINILX**
substition of class-validator to zod in dtos from users and results;
addition of nestjs-zod library to be able to add Zod Schemas into Swagger.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -7,12 +7,15 @@ import {
...
 import { AppModule } from './app.module';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
 
+  patchNestJsSwagger();
+
   app.enableVersioning({
     type: VersioningType.URI,
   });
```
Comment (yurimutti) [inline src/main.ts]: This package has been deprecated.

https://github.com/risen228/nestjs-zod#readme

**Comment by yurimutti** [comment]
Please remove nestjs-zod.

**Review by yurimutti — CHANGES_REQUESTED** [review]
Please remove nestjs-zod.

File: src/main.ts (hunk 1 above)
Comment (XVINILX) [reply src/main.ts]: removed the nestjs-zod package.

File: src/modules/user/dtos/create-user.dto.ts [hunk 2]
```diff
@@ -1,21 +1,15 @@
...
-
-  @IsArray()
-  @IsString({ each: true })
-  roleIds: string[];
-}
+import { z } from 'zod';
+
+export const CreateUserSchema = z.object({
+  email: z
+    .string({ message: 'wasteFootprint must be a string' })
```
Comment (yurimutti) [inline src/modules/user/dtos/create-user.dto.ts]: Here is email but the message is    " .string({ message: 'wasteFootprint must be a string' })"

Comment (yurimutti) [reply src/modules/user/dtos/create-user.dto.ts:5]: Correct -> email: z.string({
    message: 'email must be a string and should not be empty',
  }),

File: src/modules/calculator/dtos/result.dto.ts [hunk 3]
```diff
@@ -1,5 +1,11 @@
-export class ResultDto {
-  walletAddress: string;
-  cRecys: number;
-  wasteFootprint: number;
-}
+import { z } from 'zod';
+
+export const ResultDtoSchema = z.object({
+  walletAddress: z.string({ message: 'wasteFootprint must be a string' }),
+  cRecys: z.number({ message: 'wasteFootprint must be a number' }),
```
Comment (yurimutti) [inline src/modules/calculator/dtos/result.dto.ts]: Here is cRecys but message refer to wasteFootPrint

File: src/modules/calculator/dtos/support.dto.ts [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
+  }),
+  companyType: z.string({
+    message: 'companyType must be a string and should not be empty',
+  }),
+  employeesQuantity: z
+    .number({ message: 'wasteFootprint must be a number' })
+    .min(1, { message: 'wasteFootprint cannot be empty' }),
+  wasteFootPrint: z
+    .number({ message: 'wasteFootprint must be a number' })
+    .min(1, { message: 'wasteFootprint cannot be empty' }),
```
Comment (yurimutti) [inline src/modules/calculator/dtos/support.dto.ts]: Here too!

File: src/modules/user/dtos/create-user.dto.ts [hunk 5]
```diff
@@ -1,21 +1,15 @@
...
+    .email('Please make sure that this is e-mail is valid'),
+  name: z.string({ message: 'name must be a string' }),
+  phone: z.string({ message: 'phone must be a string' }).optional(),
+  walletAddress: z.string({ message: 'phone must be a string' }).optional(),
```
Comment (yurimutti) [inline src/modules/user/dtos/create-user.dto.ts]: walletAddress but message refer to phone.

**Comment by yurimutti** [comment]
Please @XVINILX review all strings messages.

File: src/modules/calculator/dtos/support.dto.ts (hunk 4 above)
Comment (yurimutti) [reply src/modules/calculator/dtos/support.dto.ts:15]: You can use z.email() 


email: z.string().email(),  //only this is enough

email: z.string().email({message:"Email is required"}), // use this to show custom validation message

File: src/modules/calculator/dtos/result.dto.ts [hunk 6]
```diff
@@ -1,5 +1,11 @@
-export class ResultDto {
-  walletAddress: string;
-  cRecys: number;
-  wasteFootprint: number;
-}
+import { z } from 'zod';
+
+export const ResultDtoSchema = z.object({
+  walletAddress: z.string({ message: 'walletAddress must be a string' }),
```
Comment (yurimutti) [inline src/modules/calculator/dtos/result.dto.ts]: Could you update here to validate EVM wallet ?

```
walletAddress: z
    .string()
    .regex(/^0x[a-fA-F0-9]{40}$/, 'Invalid EVM wallet address format')
    .or(z.literal(''))
    .optional(),
```

https://github.com/detrash/recy-api/blob/staging/src/modules/recycling-report/dtos/create-recycling-report.dto.ts

**Review by yurimutti — APPROVED** [review]

**Pull request merged**